### PR TITLE
Devise user management

### PR DESCRIPTION
**Pull request by timbuchwaldt**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  devise_for :users, controllers: { omniauth_callbacks: 'users/omniauth_callbacks' }
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+
+  root to: "home#index"
```
Comment (jfornoff) [inline config/routes.rb]: Where does this lead?

File: db/migrate/20200329121109_devise_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class DeviseCreateUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+
```
Comment (jfornoff) [inline db/migrate/20200329121109_devise_create_users.rb]: Maybe add a name?

File: config/routes.rb (hunk 1 above)
Comment (timbuchwaldt) [reply config/routes.rb]: Currently no where, will fix when spectest is done

File: db/migrate/20200329121109_devise_create_users.rb (hunk 2 above)
Comment (timbuchwaldt) [reply db/migrate/20200329121109_devise_create_users.rb]: Would do later on, k?

Comment (jfornoff) [reply db/migrate/20200329121109_devise_create_users.rb]: K

**Comment by jfornoff** [comment]
Pls solve merge conflicts :^)

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      set_flash_message(:notice, :success, kind: "Github") if is_navigational_format?
+    else
+      # session["devise.github_data"] = request.env["omniauth.auth"]
+      redirect_to new_user_registration_url
+    end
+  end
+
+  def failure
+    redirect_to root_path
+  end
```
Comment (jfornoff) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Install rufo :D

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,11 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
+  devise_for :users, controllers: { omniauth_callbacks: 'users/omniauth_callbacks' }
   resources :project_profiles
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+
+  root to: 'page#home'
+
+  get '/home', to: 'page#home'
```
Comment (jfornoff) [inline config/routes.rb]: Looks redundant

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -14,13 +19,23 @@
...
+    end
+  end
   # rspec-expectations config goes here. You can use an alternate
   # assertion/expectation library such as wrong or the stdlib/minitest
   # assertions if you prefer.
   config.expect_with :rspec do |expectations|
     # This option will default to `true` in RSpec 4. It makes the `description`
     # and `failure_message` of custom matchers include text for helper methods
-    # defined using `chain`, e.g.:
+    # defined using `chpain`, e.g.:
```
Comment (jfornoff) [inline spec/spec_helper.rb]: chpain huh, you mean shpain?

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -10,6 +10,10 @@
   </head>
 
   <body>
+  <div>
```
Comment (jfornoff) [inline app/views/layouts/application.html.erb]: Indent

Comment (timbuchwaldt) [reply app/views/layouts/application.html.erb]: alda

**Pull request merged**